From 3445bdbb156a8ebc2090760b9747b333437ba2ab Mon Sep 17 00:00:00 2001
From: kxxt <rsworktech@outlook.com>
Date: Tue, 23 May 2023 18:14:36 +0800
Subject: [PATCH] SystemUI: Avoid using null Context in onTuningChanged()

After switching user, we might reach onTuningChanged() before fragment
is attached, resulting in getContext() returning null, which then leads
to NPE.

Change-Id: Ia73412b3b6438bca18d09ab39db7f0657f3675a5
---

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/fragment/CollapsedStatusBarFragment.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/fragment/CollapsedStatusBarFragment.java
index d40bafb..81b61bb3 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/fragment/CollapsedStatusBarFragment.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/fragment/CollapsedStatusBarFragment.java
@@ -28,6 +28,7 @@
 import android.annotation.Nullable;
 import android.annotation.SuppressLint;
 import android.app.Fragment;
+import android.content.Context;
 import android.content.Intent;
 import android.database.ContentObserver;
 import android.os.Bundle;
@@ -141,6 +142,7 @@
     private final Executor mMainExecutor;
     private final DumpManager mDumpManager;
     private ClockController mClockController;
+    private Context mContext;
     private boolean mIsClockBlacklisted;
 
     private List<String> mBlockedIcons = new ArrayList<>();
@@ -272,6 +274,8 @@
         initEmergencyCryptkeeperText();
         initOperatorName();
         initNotificationIconArea();
+
+        mContext = getContext();
         Dependency.get(TunerService.class).addTunable(this, StatusBarIconController.ICON_HIDE_LIST);
 
         mSystemEventAnimator =
@@ -368,7 +372,7 @@
     public void onTuningChanged(String key, String newValue) {
         boolean wasClockBlacklisted = mIsClockBlacklisted;
         mIsClockBlacklisted = StatusBarIconController.getIconHideList(
-                getContext(), newValue).contains("clock");
+                mContext, newValue).contains("clock");
         if (wasClockBlacklisted && !mIsClockBlacklisted) {
             showClock(false);
         }
